lib/backend/k8s/k8s.go

```diff
@@ -499,13 +499,19 @@ func (c *KubeClient) listWorkloadEndpoints(l model.WorkloadEndpointListOptions)
 		kvp, err := c.getWorkloadEndpoint(model.WorkloadEndpointKey{
 			WorkloadID: l.WorkloadID,
 		})
+
 		if err != nil {
-			// Error getting the endpoint.
-			return nil, err
+			switch err.(type) {
+			case errors.ErrorResourceDoesNotExist:
+				return []*model.KVPair{}, nil
+			default:
+				return nil, err
+			}
 		}
+
 		if kvp == nil {
```

Comment:
Is this branch still valid?  Seems like a duplicate of the error handling above - can we remove?